lib/validation/schema.ex

```diff
@@ -7,6 +7,10 @@ defmodule Validation.Schema do
   alias Validation.Result
   alias Validation.Rule
 
+  @type t           :: %__MODULE__{val: schema_fun, meta: meta_data}
+  @typep schema_fun :: ((any) -> Result.t)
```

Comment:
You are right. Good catch. It should be `Map.t`. No worries it exists. Most modules in the elixir standard library define these types like `Map.t`, `Range.t` etc.

Will fix that.